perception/autoware_tensorrt_bevdet/src/postprocess.cu

```diff
@@ -0,0 +1,212 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include "autoware/tensorrt_bevdet/postprocess.hpp"
+
+#include <thrust/device_vector.h>
+#include <thrust/host_vector.h>
+#include <thrust/sequence.h>
+#include <thrust/sort.h>
+#include <thrust/transform.h>
+
+__device__ float sigmoid_gpu(const float x)
+{
+  return 1.0f / (1.0f + expf(-x));
+}
+
+__global__ void BEVDecodeObjectKernel(
+  const int map_size,        // 40000
+  const float score_thresh,  // 0.1
+                             //    const int nms_pre_max_size, // 4096
+  const float x_start, const float y_start, const float x_step, const float y_step,
+  const int output_h, const int output_w, const int downsample_size, const int num_class_in_task,
+  const int cls_range, const float * reg, const float * hei, const float * dim, const float * rot,
+  const float * vel, const float * cls, float * res_box, float * res_conf, int * res_cls,
+  int * res_box_num, float * rescale_factor)
+{  // According to the confidence level, the initial screening showed that there were res-box_num
+   // boxes after screening, not exceeding nms_pre_max_size 4096
+  int idx = threadIdx.x + blockDim.x * blockIdx.x;
+  if (idx >= map_size) return;
+
+  float max_score = cls[idx];  // Initialize as the first class of the task
+  int label = cls_range;       // Initialize as the first class of the task
+  for (int i = 1; i < num_class_in_task; ++i) {
+    float cur_score = cls[idx + i * map_size];
+    if (cur_score > max_score) {
+      max_score = cur_score;
+      label = i + cls_range;
+    }
+  }
+
+  int coor_x = idx % output_h;  //
+  int coor_y = idx / output_w;  //
+
+  float conf = sigmoid_gpu(max_score);  // Calculate confidence level
+  if (conf > score_thresh) {
+    int cur_valid_box_id = atomicAdd(res_box_num, 1);
+    res_box[cur_valid_box_id * kBoxBlockSize + 0] =
+      (reg[idx + 0 * map_size] + coor_x) * x_step + x_start;
+    res_box[cur_valid_box_id * kBoxBlockSize + 1] =
+      (reg[idx + 1 * map_size] + coor_y) * y_step + y_start;
+    res_box[cur_valid_box_id * kBoxBlockSize + 2] = hei[idx];
+    res_box[cur_valid_box_id * kBoxBlockSize + 3] =
+      expf(dim[idx + 0 * map_size]) * rescale_factor[label];  // nms scale
+    res_box[cur_valid_box_id * kBoxBlockSize + 4] =
+      expf(dim[idx + 1 * map_size]) * rescale_factor[label];
+    res_box[cur_valid_box_id * kBoxBlockSize + 5] =
+      expf(dim[idx + 2 * map_size]) * rescale_factor[label];
+    res_box[cur_valid_box_id * kBoxBlockSize + 6] = atan2f(rot[idx], rot[idx + map_size]);
+    res_box[cur_valid_box_id * kBoxBlockSize + 7] = vel[idx];
+    res_box[cur_valid_box_id * kBoxBlockSize + 8] = vel[idx + map_size];
+
+    res_conf[cur_valid_box_id] = conf;
+    res_cls[cur_valid_box_id] = label;
+  }
+}
+
+PostprocessGPU::PostprocessGPU(
```

Comment:
Consider placing this code within a namespace to avoid potential naming conflicts with other parts of the project.  For example: `namespace autoware { namespace tensorrt_bevdet { class PostprocessGPU { ... }; } }`